src/js/providers/html5.js

```diff
@@ -756,7 +756,8 @@ define([
             var _selectedTextTrackIndex = -1, i = 0;
             _activeCuePosition = -1;
             for (i; i < _videotag.textTracks.length; i++) {
-                if (_videotag.textTracks[i].mode === 'showing') {
+                if ((_videotag.textTracks[i].kind === 'subtitles' || _videotag.textTracks[i].kind === 'captions')
+                    && _videotag.textTracks[i].mode === 'showing') {
```

Comment:
You can use _.findIndex to get the selected text track index. And you only need to do it when a selected track is found.